cloud/gcpprovider.go

```diff
@@ -560,11 +578,10 @@ func (gcp *GCP) parsePage(r io.Reader, inputKeys map[string]Key, pvKeys map[stri
 									product.Node = &Node{
 										RAMCost: strconv.FormatFloat(hourlyPrice, 'f', -1, 64),
 									}
-									/*
-										if partialCPU != 0 {
-											product.Node.VCPU = fmt.Sprintf("%f", partialCPU)
-										}
-									*/
+									partialCPU, pcok := partialCPUMap[instanceType]
+									if pcok {
+										product.Node.VCPU = fmt.Sprintf("%f", partialCPU)
```

Comment:
It's pretty tough to tell what's going on here for such a big and deeply-nested function, so I just want to check: this block appears to be setting RAM cost for the given node. Do we also want to be setting VCPU here? Doesn't seem harmful, I'm just confused.

Reply:
Agreed its confusing, but the pattern is (or should be): anytime you add a new object to the block, set its VCPU, since we know how many virtualCPUs there are as soon as we know the key.